scripts/sct_get_centerline.py

```diff
@@ -4,135 +4,157 @@
-                      mandatory=False,
-                      default_value='20.0')
-    parser.add_option(name="-igt",
-                      type_value="image_nifti",
-                      description="File name of ground-truth centerline or segmentation (binary nifti).",
-                      mandatory=False)
-    parser.add_option(name="-v",
-                      type_value="multiple_choice",
-                      description="1: display on, 0: display off (default)",
-                      mandatory=False,
-                      example=["0", "1", "2"],
-                      default_value="1")
-    parser.add_option(name='-qc',
-                      type_value='folder_creation',
-                      description='The path where the quality control generated content will be saved',
-                      default_value=None)
-    parser.add_option(name='-qc-dataset',
-                      type_value='str',
-                      description='If provided, this string will be mentioned in the QC report as the dataset the process was run on',
-                      )
-    parser.add_option(name='-qc-subject',
-                      type_value='str',
-                      description='If provided, this string will be mentioned in the QC report as the subject the process was run on',
-                      )
+    parser = argparse.ArgumentParser(
+        description=(
+            "This function extracts the spinal cord centerline. Three methods are available: OptiC (automatic), "
+            "Viewer (manual), and Fitseg (applied on segmented image). These functions output (i) a NIFTI file with "
+            "labels corresponding to the discrete centerline, and (ii) a csv file containing the float (more precise) "
+            "coordinates of the centerline in the RPI orientation. \n"
+            "\n"
+            "Reference: C Gros, B De Leener, et al. Automatic spinal cord localization, robust to MRI contrast using "
+            "global curve optimization (2017). doi.org/10.1016/j.media.2017.12.001"
+        ),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Input image. Example: ti.nii.gz"
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        "-c",
+        choices=['t1', 't2', 't2s', 'dwi'],
+        help="Type of image contrast. Only with method=optic."
+    )
+    optional.add_argument(
+        "-method",
+        choices=['optic', 'viewer', 'fitseg'],
+        default='optic',
+        help=("Method used for extracting the centerline.\n"
```

Comment:
in my terminal, this displays as:
~~~
  -method {optic,viewer,fitseg}
                        Method used for extracting the centerline. - optic: automatic spinal cord
                        detection method - viewer: manual selection a few points followed by
                        interpolation - fitseg: fit a regularized centerline on an already-existing
                        cord segmentation. It will interpolate if slices are missing and extrapolate
                        beyond the segmentation boundaries (i.e., every axial slice will exhibit a
                        centerline pixel).
~~~